pysat/_meta.py

```diff
@@ -1021,21 +1021,99 @@ def attr_case_name(self, name):
         variable name.
 
         """
+        # Get a case-insensitive version of the name
         lower_name = name.lower()
+
+        # Determine if the input name is one of the top-level attributes
         for out_name in self.attrs():
             if lower_name == out_name.lower():
                 return out_name
 
-        # check if attribute present in higher order structures
+        # Determine if the attribute is present in higher order structures
         for key in self.keys_nD():
             for out_name in self[key].children.attrs():
                 if lower_name == out_name.lower():
                     return out_name
 
-        # nothing was found if still here
-        # pass name back, free to be whatever
+        # Nothing was found if still here. Pass `name` back without alteration,
+        # it is free to be whatever it is.
         return name
 
+    def rename(self, mapper):
+        """Update the preserved case name for case insensitive value of name.
+
+        Parameters
+        ----------
+        mapper : dict or func
+            Dictionary with old names as keys and new names as variables or
+            a function to apply to all names
+
+        Note
+        ----
+        Checks first within standard attributes. If not found there, checks
+        attributes for higher order data structures. If not found, returns
+        supplied name as it is available for use. Intended to be used to help
+        ensure that the same case is applied to all repetitions of a given
+        variable name.
+
+        """
+        def get_mapped_value(value, mapper):
```

Comment:
Ok, so I went to go do this and ran into the problem that this method doesn't use `self` at all.  So, I could put it as a hidden `_meta` method, keep it here, or I could put it somewhere in `utils`.  Not sure where it would go besides `_core`.  But `utils` may be the right place if we want to eventually add a `rename` method to `Instrument`.

Reply:
Good point.

> Ok, so I went to go do this and ran into the problem that this method doesn't use `self` at all. 

Ah, right...

> So, I could put it as a hidden `_meta` method, keep it here, or I could put it somewhere in `utils`. 

Given that technical point I think where it is is fine then.

> if we want to eventually add a `rename` method to `Instrument`.

We got one :)

Reply:
> We got one :)

Ok, if we ever want to *upgrade* Instrument.rename. 😜

Reply:
Made an issue to track that there is now technically two code paths in the repo for changing a metadata variable name. pysat.Instrument has its own code for renaming metavariables that should be updated to use meta.rename. #900